### DISCUSSION TITLE
Using Z2M as Docker Container with Silabs Add-On in Home Assistant

### DISCUSSION THREAD

**Discussion by GerdRuetten** [Koenkk/zigbee2mqtt#21028] [General]
Hello community,

Until last week I used z2m outside of Home Assistant in its own Docker container with a ZBDongle-P stick.
Everything problem-free...

Now I have switched to a ZBDongle-E stick with multi-protocol firmware.
In HA I use the add-on from SiLabs and so far Z2M again as an HA add-on.
This also works without errors.

But I would like to run Z2M outside of HA in a separate container again (like before).

However, I am unable to configure the serial parameters correctly in the Z2M configuration.

Within HA, i.e. as an add-on it works with:
port: tcp://core-silabs-multiprotocol:9999
adapter: ezsp

But what do I have to specify under “Port” if I no longer have Z2M running as an add-on?
Is this even possible (currently)?

Thanks for any support!

Best regards
Gerd

**Comment by p0wertiger** [comment]
Technically speaking, USB dongle is a serial device so it shoud be visible as /dev/ttyUSB01 or /dev/ttyACM01 or similar. I don't know about multiprotocol firmware, I've been wondering how to make it work as well (although I will move to Ember FW), most probably run the SiLabs addon code in a container and then connect Z2M to it.
To pass device control to container, use `--device` Docker option

**Reply by p0wertiger** [reply]
You can test the add-on locally by getting it [here](https://github.com/home-assistant/addons/tree/master/silabs-multiprotocol) and following [these steps](https://developers.home-assistant.io/docs/add-ons/testing) .
